services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -0,0 +1,412 @@
+    for feature_name, feature in tqdm(categorical_features.items()):
+        logging.debug(f"Compute statistics for ClassLabel feature {feature_name}")
+        class_label_names = feature["names"]
+        cat_column_stats: CategoricalStatisticsItem = compute_categorical_statistics(
+            con,
+            feature_name,
+            class_label_names=class_label_names,
+            n_samples=num_examples,
+            parquet_filename=local_parquet_path,
+        )
+        stats.append(
+            StatisticsPerColumnItem(
+                column_name=feature_name,
+                column_type=ColumnType.CLASS_LABEL,
+                column_statistics=cat_column_stats,
+            )
+        )
+
+    if numerical_features:
+        logging.info(f"Compute min, max, mean, median, std, histogram for numerical columns {numerical_features}. ")
+    for feature_name, feature in tqdm(numerical_features.items()):
+        column_type = ColumnType.FLOAT if feature["dtype"] in FLOAT_DTYPES else ColumnType.INT
+        num_column_stats: NumericalStatisticsItem = compute_numerical_statistics(
+            con,
+            feature_name,
+            parquet_filename=local_parquet_path,
+            n_bins=histogram_num_bins,
+            n_samples=num_examples,
+            column_type=column_type,
+        )
+        stats.append(
+            StatisticsPerColumnItem(
+                column_name=feature_name,
+                column_type=column_type,
+                column_statistics=num_column_stats,
+            )
+        )
+    con.close()
+
+    return SplitDescriptiveStatisticsResponse(
+        num_examples=num_examples, statistics=sorted(stats, key=lambda x: x["column_name"])
+    )
+
+
+class SplitDescriptiveStatisticsJobRunner(SplitJobRunnerWithCache):
+    descriptive_statistics_config: DescriptiveStatisticsConfig
+
+    def __init__(
+        self,
+        job_info: JobInfo,
+        app_config: AppConfig,
+        processing_step: ProcessingStep,
+        statistics_cache_directory: StrPath,
+    ):
+        super().__init__(
+            job_info=job_info,
+            app_config=app_config,
+            processing_step=processing_step,
+            cache_directory=Path(statistics_cache_directory),
+        )
+        self.descriptive_statistics_config = app_config.descriptive_statistics
+
+    @staticmethod
+    def get_job_type() -> str:
+        return "split-descriptive-statistics"
+
+    @staticmethod
+    def get_job_runner_version() -> int:
+        return PROCESSING_STEP_SPLIT_DESCRIPTIVE_STATISTICS_VERSION
+
+    def compute(self) -> CompleteJobResult:
+        return CompleteJobResult(
+            compute_descriptive_statistics_response(
+                dataset=self.dataset,
+                config=self.config,
+                split=self.split,
+                local_parquet_directory=self.cache_subdirectory,
+                extensions_directory=self.descriptive_statistics_config.extensions_directory,
```

Comment:
Maybe extensions_directory can be a path inside local_parquet_directory ?
This way no need to define a second directory, and avoid having the optional parameter `extensions_directory `

Reply:
sounds good. separate `extensions_directory` came from `split-duckdb-index` but maybe it's really not needed, @AndreaFrancis ?

Reply:
I think in your case it could be local_parquet_directory. In `split-duckdb-index` I didn't have another alternative folder (besides duckdb_index_file_directory root).